src/actions/userStat/getUserStat.js

```diff
@@ -0,0 +1,165 @@
+  return newFollowers.length;
+};
+
+const fetchData = async (url, token) => {
+  axios.defaults.headers.common = {
+    Authorization: `bearer ${token}`,
+  };
+  const response = await axios.get(url);
+  return response;
+};
+
+export
+const userFollowerStat = (token, userId) => async (dispatch) => {
+  try {
+    const url = `${process.env.SERVER_API}/followers/${userId}`;
+    const response = await fetchData(url, token);
+    const followersStats = {
+      newFollowers: getNewFollowers(response.data.followers.rows),
+      totalFollowers: response.data.followers.count,
+    };
+    return dispatch({
+      type: actionTypes.GET_FOLLOWERS_SUCCESS,
+      payload: followersStats,
+    });
+  } catch (error) {
+    return dispatch({
+      type: actionTypes.GET_FOLLOWERS_FAILURE,
+      error,
+    });
+  }
+};
+
+export
+const followingUsersStat = (token, userId) => async (dispatch) => {
+  try {
+    const url = `${process.env.SERVER_API}/following/${userId}`;
+    const response = await fetchData(url, token);
+    return dispatch({
+      type: actionTypes.FOLLOWING_OTHERS_STAT_SUCCESS,
+      payload: !response.data.followees ? 0 : response.data.followees.length,
+    });
+  } catch (error) {
+    return dispatch({
+      type: actionTypes.FOLLOWING_OTHERS_STAT_FAILURE,
+      error,
+    });
+  }
+};
+
+
+export
+const userCommentStat = token => async (dispatch) => {
+  const url = `${process.env.SERVER_API}/comments/user/count`;
+  try {
+    const response = await fetchData(url, token);
+    const commentStats = response.data.data.count;
+    return dispatch({
+      type: actionTypes.GET_COMMENT_STAT_SUCCESS,
+      payload: commentStats,
+    });
+  } catch (error) {
+    return dispatch({
+      type: actionTypes.GET_COMMENT_STAT_FAILURE,
+      error,
+    });
+  }
+};
+
+export
+const userArticleLikeStat = token => async (dispatch) => {
+  const url = `${process.env.SERVER_API}/votes/user`;
+  try {
+    const response = await fetchData(url, token);
+    const likesStats = response.data.data.count;
+    return dispatch({
+      type: actionTypes.GET_LIKED_ARTICLES_SUCCESS,
+      payload: likesStats,
+    });
+  } catch (error) {
+    return dispatch({
+      type: actionTypes.GET_LIKED_ARTICLES_FAILURE,
+      error,
+    });
+  }
+};
+
+export
+const userBookmarkStat = token => async (dispatch) => {
+  const url = `${process.env.SERVER_API}/bookmark`;
+  try {
+    const response = await fetchData(url, token);
+    const result = response.data.data.count;
+    return dispatch({
+      type: actionTypes.GET_BOOKMARK_SUCCESS,
+      payload: result,
+    });
+  } catch (error) {
+    return dispatch({
+      type: actionTypes.GET_BOOKMARK_FAILURE,
+      error,
+    });
+  }
+};
+
+const calTotal = (array, props) => array.reduce((a, b) => a + b[props], 0);
+
+const articleStats = (articleData) => {
+  const userArticleStats = {
+    noOfreaders: calTotal(articleData.rows, 'readingStat'),
```

Comment:
Hi @JohnMadakin. Nice job done here. But I feel you could have used `numberOfReaders` instead.

Reply:
done